api/v1alpha1/authenticationfilter_types.go

```diff
@@ -119,6 +128,58 @@ type RemoteJWKS struct {
 	// TODO: Add TBD remote JWKS fields based on defined use cases.
 }
 
+// OIDCAuthenticationFilterProvider defines the OpenID Connect (OIDC) authentication provider type
+type OIDCAuthenticationFilterProvider struct {
+	// The OIDC Provider configuration.
+	Provider OIDCProvider `json:"provider"`
+
+	// The OIDC client ID assigned to the filter to be used in the
+	// [Authentication Request](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest).
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	ClientID string `json:"clientId"`
+
+	// The name of the Kubernetes secret which contains the OIDC client secret assigned to the filter to be used in the
+	// [Authentication Request](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest).
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	ClientSecret string `json:"clientSecret"`
```

Comment:
can we leverage the same style as upstream for representing a secret https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1beta1.GatewayTLSConfig